sled-agent/src/services.rs

```diff
@@ -725,6 +739,42 @@ impl ServiceManager {
         )
         .await?;
 
+        // TODO(https://github.com/oxidecomputer/omicron/issues/1898):
+        //
+        // These zones are self-assembling -- after they boot, there should
+        // be no "zlogin" necessary to initialize.
+        match request.zone_type {
+            ZoneType::CruciblePantry => {
+                let Some(info) = self.inner.sled_info.get() else {
+                    return Err(Error::SledAgentNotReady);
+                };
+
+                let datalink = installed_zone.get_control_vnic_name();
+                let gateway = &info.underlay_address.to_string();
+                assert_eq!(request.addresses.len(), 1);
+                let listen_addr = &request.addresses[0].to_string();
+                let listen_port = &CRUCIBLE_PANTRY_PORT.to_string();
+
+                let config = PropertyGroupBuilder::new("config")
+                    .add_property("datalink", "astring", datalink)
+                    .add_property("gateway", "astring", gateway)
+                    .add_property("listen_addr", "astring", listen_addr)
+                    .add_property("listen_port", "astring", listen_port);
+
+                let profile = ProfileBuilder::new("omicron").add_service(
+                    ServiceBuilder::new("system/illumos/crucible_pantry")
```

Comment:
This should really be `oxide/crucible/pantry`, rather than `system/illumos/crucible_pantry`.

Reply:
Renamed